bitnami/etcd/templates/scripts-configmap.yaml

```diff
@@ -0,0 +1,184 @@
+        exit 0
+    }
+    ## Configure RBAC
+    configure_rbac() {
+        # When there's more than one replica, we can assume the 1st member
+        # to be created is "{{ $etcdFullname }}-0" since a statefulset is used
+        if [[ -n "$ETCD_ROOT_PASSWORD" ]] && [[ "$HOSTNAME" == "{{ $etcdFullname }}-0" ]]; then
+            echo "==> Configuring RBAC authentication!"
+            etcd > /dev/null 2>&1 &
+            ETCD_PID=$!
+            while ! etcdctl member list >/dev/null 2>&1; do sleep 1; done
+            echo "$ETCD_ROOT_PASSWORD" | etcdctl user add root --interactive=false
+            etcdctl auth enable
+            kill "$ETCD_PID"
+            sleep 5
+        fi
+    }
+    ## Checks whether there was a disaster or not
+    is_disastrous_failure() {
+        local endpoints_array=(${ETCDCTL_ENDPOINTS//,/ })
+        local active_endpoints=0
+        local -r min_endpoints=$((({{ $replicaCount }} + 1)/2))
+
+        for e in "${endpoints_array[@]}"; do
+            if [[ "$e" != "$ETCD_ADVERTISE_CLIENT_URLS" ]] && etcdctl $AUTH_OPTIONS  endpoint health --endpoints="$e" >/dev/null 2>&1; then
+                active_endpoints=$((active_endpoints + 1))
+            fi
+        done
+
+        if [[ -f "/snapshots/.disaster_recovery" ]]; then
+            if [[ $active_endpoints -eq $(({{ $replicaCount }} - 1)) ]]; then
+                echo "==> I'm the last to recover from the disaster!"
+                rm "/snapshots/.disaster_recovery"
+            fi
+            true
+        else
+            if [[ $active_endpoints -lt $min_endpoints ]]; then
+                touch "/snapshots/.disaster_recovery"
+                true
+            else
+                false
+            fi
+        fi
+    }
+
+    if [[ ! -d "$ETCD_DATA_DIR" ]]; then
+        echo "==> Creating data dir..."
+        mkdir -p "$ETCD_DATA_DIR"
+        echo "==> There is no data at all. Initializing a new member of the cluster..."
+        store_member_id &
+        configure_rbac
+    else
+        echo "==> Detected data from previous deployments..."
+        if is_disastrous_failure; then
+            echo "==> Cluster not responding!!"
+{{- if .Values.disasterRecovery.enabled }}
+            if [[ -f "/snapshots/db" ]]; then
+                echo "==> Restoring etcd cluster from snapshot..."
+
+                rm -rf $ETCD_DATA_DIR
+                etcdctl snapshot restore /snapshots/db \
+                  --name $ETCD_NAME \
+                  --data-dir $ETCD_DATA_DIR \
+                  --initial-cluster $ETCD_INITIAL_CLUSTER \
+                  --initial-cluster-token $ETCD_INITIAL_CLUSTER_TOKEN \
+                  --initial-advertise-peer-urls $ETCD_INITIAL_ADVERTISE_PEER_URLS
+                store_member_id &
+            else
+                echo "==> There was no snapshot to perform data recovery!!"
+                exit 1
+            fi
+{{- else }}
+            exit 1
```

Comment:
Should we print here that disasterRecovery was disabled and so we cannot recover the cluster?

Reply:
done